spec/models/benefit_group_spec.rb

```diff
@@ -613,35 +613,51 @@
   end
 
   describe BenefitGroup, dbclean: :after_each do
-    let(:benefit_group) { FactoryGirl.create(:benefit_group)}
-    let(:carrier_profile) { FactoryGirl.create(:carrier_profile) }
-    let!(:plan) { FactoryGirl.create(:plan, carrier_profile: carrier_profile)}
-    context "plan with hios id 34484" do
-       before do
-        allow(carrier_profile).to receive(:is_group_size?).and_return true
-       end
+    let(:benefit_group) { FactoryGirl.create(:benefit_group, plan_option_kind: 'sole_source', reference_plan: plan)}
+    let(:default_benefit_group) { FactoryGirl.create(:benefit_group, plan_option_kind: 'sole_source', reference_plan: default_carrier_plan)}
+    let(:carrier_profile_one) { FactoryGirl.create(:carrier_profile, issuer_hios_ids: ['11111']) }
+    let(:composite_carrier_two) { FactoryGirl.create(:carrier_profile, issuer_hios_ids: ['22222']) }
+    let(:number_of_employees) { 1 }
+    let!(:plan) { FactoryGirl.create(:plan, carrier_profile: carrier_profile_one)}
+    let!(:default_carrier_plan) { FactoryGirl.create(:plan, carrier_profile: composite_carrier_two)}
+
+    let(:carrier_one_size_3) { build(:rating_factor_entry, factor_key: 3, factor_value: 1.101) }
```

Comment:
See how we build each factor entry, then associated it with the factor_set that is associated to the carrier -> plan -> benefit_group.

This way when we ask benefit_group what the group_size_factor_for is it finds the correct set and returns it.